#! /usr/bin/env python
#
# Copyright (c) 2025 Mark A. Bowers
# All rights reserved.
#
# IGVC Perception Project
# https://github.com/mabowers/igvc-perception
# ECE5532 Autonomous Vehicle Systems I, Winter 2025
# Oakland University, Rochester, MI
# Instructor: Dr. Micho Radovnikovich
#
# Dynamic reconfiguration parameters for the CameraProcessing node
#
# This file is based on the following example from the course materials;
# https://github.com/robustify/winter2025_ece5532_examples/blob/master/opencv_example/cfg/HoughTransform.cfg
#

PACKAGE='igvc_perception' # Name of package here

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name                    Type      Level Description                             Default  Min    Max
gen.add("blue_thres",           int_t,    0,    "Threshold",                            200,     0,     255)
gen.add("erode_size",           int_t,    0,    "Erode Size",                           5,       1,     21)
gen.add("dilate_size",          int_t,    0,    "Dilate Size",                          5,       1,     21)
gen.add("hough_rho_res",        int_t,    0,    "Rho resolution (pix)",                 10,      1,     500)
gen.add("hough_theta_res",      double_t, 0,    "Theta resolution (rad)",               0.05,    0.01,  1.0)
gen.add("hough_threshold",      int_t,    0,    "Accumulator threshold",                10,      10,    200)
gen.add("hough_min_length",     int_t,    0,    "Minimum line length (pix)",            100,     5,     500)
gen.add("hough_max_gap",        int_t,    0,    "Max gap betweeen line segments (pix)", 50,      5,     500)
gen.add("frame_jitter",         int_t,    0,    "Pre-processed frame jitter",           1,      -5,     5)

exit(gen.generate(PACKAGE, PACKAGE, "CameraProcessing"))
